src/main/java/emissary/util/ObjectTracing.java

```diff
@@ -0,0 +1,38 @@
+package emissary.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ObjectTracing {
+
+    private ObjectTracing() {}
+
+    private static final List<String> fieldNames = new ArrayList<>();
+
+    protected static final Logger logger = LoggerFactory.getLogger(ObjectTracing.class);
+
+    public static void setUpFieldNames(List<String> newFieldNames) {
+        fieldNames.addAll(newFieldNames);
+    }
+
+    public static Map<String, String> createTraceMessageMap(String[] fieldValues) {
+
+        Map<String, String> jsonMap = new HashMap<>();
+
+        if (fieldValues.length != fieldNames.size()) {
+            logger.error("Cannot create log entry, the number of fields does not equals the number of values");
+            return Collections.emptyMap();
+        }
+
+        for (int i = 0; i < fieldValues.length; i++) {
```

Comment:
I see what you are getting at with making the fields configurable and then iterating through the array to make the json map.

But from a performance and complexity stand point do we really need it to be configurable?

Say in this util class could we just have a method that is buildPickupMessage(String filename)
and build the message we need with the fields we need and have it all in code. 
That method can be overridden in other software components instead of a config file.

I like the idea of having the configuration but worried that the fields need to be kept in sync in multiple places. And the order that they are added maters to keep key's and values from getting mixed up.

I don't have a better solution. Just trying to think out loud here.

Reply:
You have a good point. If we don't need the fields in a config file, I'll play around with another solution.

Reply:
I tested it and your suggestion of having a method to be overridden worked out well, so I went with that solution instead of a config file for the fields.